components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java

```diff
@@ -215,6 +218,62 @@ private static void exportAPIThumbnail(String archivePath, APIIdentifier apiIden
+     *                            storing in the archive directory
+     */
+    private static void exportSOAPToRESTMediation(String archivePath,APIIdentifier apiIdentifier, Registry registry)
+            throws APIImportExportException {
+        String soapToRestBaseUrl = "/apimgt/applicationdata/provider" + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getProviderName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getApiName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getVersion() + RegistryConstants.PATH_SEPARATOR +
+                "soap_to_rest";
+
+        InputStream inputStream = null;
+        OutputStream outputStream = null;
+        try {
+            if (registry.resourceExists(soapToRestBaseUrl)) {
+                Collection inFlow = (org.wso2.carbon.registry.api.Collection)registry.get(soapToRestBaseUrl + RegistryConstants.PATH_SEPARATOR + "in");
+                Collection outFlow = (org.wso2.carbon.registry.api.Collection)registry.get(soapToRestBaseUrl + RegistryConstants.PATH_SEPARATOR + "out");
+
+                createDirectory(archivePath + File.separator + "SoapToRest/in");
+                createDirectory(archivePath + File.separator + "SoapToRest/out");
+                if (inFlow != null) {
+                    for (String inFlowPath : inFlow.getChildren()) {
+                        inputStream = registry.get(inFlowPath).getContentStream();
+                        outputStream = new FileOutputStream(archivePath + File.separator + "SoapToRest" + File.separator + "in" +
+                                inFlowPath.substring(inFlowPath.lastIndexOf(RegistryConstants.PATH_SEPARATOR)));
+                        IOUtils.copy(inputStream, outputStream);
+                    }
+                }
+                if (outFlow != null) {
+                    for (String outFlowPath : outFlow.getChildren()) {
+                        inputStream = registry.get(outFlowPath).getContentStream();
+                        outputStream = new FileOutputStream(archivePath + File.separator + "SoapToRest" + File.separator + "out" +
+                                outFlowPath.substring(outFlowPath.lastIndexOf(RegistryConstants.PATH_SEPARATOR)));
+                        IOUtils.copy(inputStream, outputStream);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            log.error("I/O error while  writing API SOAP to REST logic to file", e);
+            throw new APIImportExportException("I/O error while writing API SOAP to REST logic to file", e);
+        } catch (RegistryException e) {
+            log.error("Error while retrieving API SOAP to REST logic ", e);
+            throw new APIImportExportException("Error while retrieving SOAP to REST logic", e);
+        } finally {
+            IOUtils.closeQuietly(inputStream);
```

Comment:
We can use try-with resources

Reply:
inputStream isn't a final variable, therefore, we can't use as a try-with resource